source/agora/consensus/EnrollmentManager.d

```diff
@@ -184,11 +184,46 @@ public class EnrollmentManager
 
     ***************************************************************************/
 
-    public Height getEnrolledHeight (const ref Hash enroll_hash) @trusted
+    public Height getEnrolledHeight (const ref Hash enroll_hash)
+        @trusted nothrow
     {
         return this.validator_set.getEnrolledHeight(enroll_hash);
     }
 
+    /***************************************************************************
+
+        Get the unregistered enrollments that can be validator in the next
+        block based on the current block height.
+
+        Params:
+            enrolls = will contain the unregistered enrollments data if found
+            height = current block height
+
+        Returns:
+            The unregistered enrollments data
+
+    ***************************************************************************/
+
+    public Enrollment[] getEnrollments (ref Enrollment[] enrolls, Height height)
+        @trusted nothrow
+    {
+        enrolls.length = 0;
+        assumeSafeAppend(enrolls);
+
+        static Enrollment[] pool_enrolls;
+        this.enroll_pool.getEnrollments(pool_enrolls);
+        foreach (enroll; pool_enrolls)
+        {
+            const enroll_height = this.getEnrolledHeight(enroll.utxo_key);
+            if (enroll_height == ulong.max ||
+                height == enroll_height + this.params.ValidatorCycle - 1)
```

Comment:
```suggestion
            if (enroll_height == ulong.max ||
                height >= enroll_height + this.params.ValidatorCycle - 1)
```

If you check for an exact match, and for whatever reason the enrollment can not be added on the exact height, it will not be proposed again.